packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/ExportAllTasks.tsx

```diff
@@ -0,0 +1,188 @@
+)
+
+const BUTTON_CLASSES =
+  'flex items-center gap-2 rounded-full border border-solid border-slate-400 px-5 py-2 text-center font-sans text-sm font-semibold'
+
+const integrationToServiceName: Record<TaskServiceEnum, string> = {
+  PARABOL: 'Parabol',
+  github: Providers.GITHUB_NAME,
+  jira: 'Jira',
+  jiraServer: Providers.JIRA_SERVER_NAME,
+  azureDevOps: Providers.AZUREDEVOPS_NAME,
+  gitlab: Providers.GITHUB_NAME
+}
+
+interface Props {
+  meetingRef: ExportAllTasks_meeting$key
+}
+
+const ExportAllTasks = (props: Props) => {
+  const {meetingRef} = props
+  const mutationProps = useMutationProps()
+  const atmosphere = useAtmosphere()
+  const {onCompleted, onError, submitMutation, submitting} = mutationProps
+  const {togglePortal, originRef, menuPortal, menuProps, loadingWidth, loadingDelay} = useMenu(
+    MenuPosition.UPPER_RIGHT,
+    {
+      loadingWidth: 200
+    }
+  )
+  const meeting = useFragment(
+    graphql`
+      fragment ExportAllTasks_meeting on NewMeeting {
+        id
+        meetingType
+        teamId
+        ...ExportAllTasksMenuRoot_meeting
+        ... on RetrospectiveMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+        ... on ActionMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+      }
+    `,
+    meetingRef
+  )
+
+  // :TODO: (jmtaber129): Get this information from the tasks integration.
+  const [pushedIntegrationLabel, setPushedIntegrationLabel] = useState<string | null>(null)
+
+  const {id: meetingId, meetingType, tasks, teamId} = meeting
+  if (!tasks) {
+    return null
+  }
+
+  const filteredTasks = tasks.filter((task) => !task.integration)
+  const taskServices = Array.from(
+    tasks.reduce(
+      (serviceSet, task) => (task.taskService ? serviceSet.add(task.taskService) : serviceSet),
+      new Set<string>()
+    )
+  )
+
+  const handlePushToIntegration = (
+    integrationRepoId: string,
+    integrationProviderService: Exclude<TaskServiceEnum, 'PARABOL'>,
+    integrationLabel?: string
+  ) => {
+    submitMutation()
+    Promise.allSettled(
+      filteredTasks?.map((task) => {
+        return new Promise<void>((resolve, reject) => {
+          const variables = {
+            integrationRepoId,
+            taskId: task.id,
+            integrationProviderService: integrationProviderService
+          }
+          CreateTaskIntegrationMutation(atmosphere, variables, {
+            onError: reject,
+            onCompleted: (res, errors) => {
+              const error = getOnCompletedError(res, errors)
+              if (error) {
+                reject(error)
+              } else {
+                resolve()
+              }
+            }
+          })
+        })
+      })
+    ).then((results) => {
```

Comment:
+1 i haven't sene one of these in awhile! why not async/await?

Reply:
> why not async/await?

Because I was mentally in promise-mode when I wrote this 😅 Switched this to use async/await for the result of `Promise.allSettled`